src/Sulu/Bundle/AdminBundle/Resources/js/components/Email/tests/Email.test.js

```diff
@@ -1,46 +1,63 @@
 // @flow
 import React from 'react';
-import {render, mount} from 'enzyme';
+import {fireEvent, render, screen} from '@testing-library/react';
 import Email from '../Email';
 
 test('Email should render', () => {
     const onChange = jest.fn();
-    expect(render(<Email onChange={onChange} value={null} />)).toMatchSnapshot();
+    const {container} = render(<Email onChange={onChange} value={null} />);
+
+    expect(container).toMatchSnapshot();
 });
 
 test('Email should render with placeholder', () => {
     const onChange = jest.fn();
-    expect(render(<Email onChange={onChange} placeholder="My placeholder" value={null} />)).toMatchSnapshot();
+    const {container} = render(<Email onChange={onChange} placeholder="My placeholder" value={null} />);
+
+    expect(container).toMatchSnapshot();
 });
 
 test('Email should render with value', () => {
     const onChange = jest.fn();
     const value = 'test@test.com';
-    expect(render(<Email onChange={onChange} value={value} />)).toMatchSnapshot();
+    const {container} = render(<Email onChange={onChange} value={value} />);
+
+    expect(container).toMatchSnapshot();
 });
 
 test('Email should render when disabled', () => {
     const onChange = jest.fn();
     const value = 'test@test.com';
-    expect(render(<Email disabled={true} onChange={onChange} value={value} />)).toMatchSnapshot();
+    const {container} = render(<Email disabled={true} onChange={onChange} value={value} />);
+
+    expect(container).toMatchSnapshot();
 });
 
 test('Email should render null value as empty string', () => {
     const onChange = jest.fn();
-    expect(render(<Email onChange={onChange} value={null} />)).toMatchSnapshot();
+    const {container} = render(<Email onChange={onChange} value={null} />);
+
+    expect(container).toMatchSnapshot();
 });
 
 test('Email should render error', () => {
     const onChange = jest.fn();
-    expect(render(<Email onChange={onChange} valid={false} value={null} />)).toMatchSnapshot();
+    const {container} = render(<Email onChange={onChange} valid={false} value={null} />);
+
+    expect(container).toMatchSnapshot();
 });
 
 test('Email should not set onIconClick when value is invalid', () => {
     const onChange = jest.fn();
     const onBlur = jest.fn();
-    const email = mount(<Email onBlur={onBlur} onChange={onChange} valid={false} value={null} />);
+    const onIconClickSpy = jest.fn();
+
+    render(<Email onBlur={onBlur} onChange={onChange} onIconClick={onIconClickSpy} valid={false} value={null} />);
```

Comment:
This is not possible as `Email` does not have a `OnIconClick` and that is why `Node Lint` task is failing.

Maybe we can check that nothing is changed on window.location.assign when icon is clicked. Or we can check if const icon does not have a `onClick` registered on its dom element.

Reply:
Yes, that's a good idea.